### ISSUE TITLE
feat: apr finetune should emit structured training metrics JSON (loss, throughput, VRAM)

### ISSUE THREAD

**Issue by noahgift** [paiml/aprender#566]
## Context

qwen-train-canary runs `apr finetune` as a canary to benchmark entrenar training 
throughput against unsloth/pytorch. The canary wrapper needs training metrics:
- final_loss (mean of last N steps)
- tokens_per_sec 
- peak_vram_mb
- step_time_ms (per-step timing)
- loss trajectory (per-step losses)

## Current State

`apr finetune --json` only emits planning output (memory breakdown, recommendations)
as JSON to stdout. All training output goes to stderr as unstructured trace logs:

```
[TRACE] dequant n=1536 k=1536 ...
[CUDA] Uploaded 28 transformer layers ...
[CUDA] Skipping GPU embeddings ...
```

No per-step loss, no final_loss, no throughput metrics are emitted.

The canary wrapper can only estimate throughput from wall_time 
(43 tok/s = 50 samples * 512 seq / 595s) and gets 0 for loss and VRAM.

## Requested

When `--json` is passed, emit a training summary to stdout after training completes:

```json
{
  "training": {
    "steps": 20,
    "final_loss": 3.27,
    "tokens_per_sec": 43.0,
    "peak_vram_mb": 7138,
    "wall_time_sec": 595.2,
    "loss_trajectory": [4.86, 4.50, 4.12, ..., 3.27]
  }
}
```

This would allow the canary wrapper to extract real metrics instead of estimates.

## Impact

Without structured metrics, the APR canary result quality is poor:
- final_loss: 0 (unknown)
- peak_vram_mb: 0 (unknown)  
- throughput_samples_sec: 0 (unknown)
- Only tokens_per_sec has a value (estimated from wall_time)

Refs: PMAT-454, paiml/qwen-train-canary#12
